crates/bin/cli-wallet/src/main.rs

```diff
@@ -545,6 +560,45 @@ async fn main() -> Result<()> {
         Commands::Sync => {
             sync::sync_all_pending_operations(pool).await?;
         }
+        Commands::Init {  } => {
+            let seed_phrase = wallet::utils::create_seed_phrase().unwrap();
+            println!("Seed phrase: {}", seed_phrase.to_string());
+
+            let private_key = wallet::utils::derive_private_key(&seed_phrase).unwrap();
+
+            let wallet = wallet::db::wallet::Wallet {
+                seed_phrase: seed_phrase.to_string(),
+                private_key: private_key.to_string(),
+                is_user_saved_locally: true,
+                created_at: std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_secs() as u64,
+                updated_at: std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH).unwrap().as_secs() as u64,
+            };
+
+            println!("Do you want to save this wallet locally? (y/YES/n): ");
+            let mut input = String::new();
+            std::io::stdin().read_line(&mut input)?;
+
+            let should_save = input.trim().to_lowercase();
+            if should_save == "y" || should_save == "yes" {
+                wallet::db::wallet::create_wallet(&db_conn, wallet)?;
+                println!("Wallet saved locally!");
+            } else {
+                println!("Wallet not saved locally and in Database. Please generate a new wallet and type yes/y.");
+            }
+        }
+        Commands::Restore {
+            seed_phrase,
+        } => {
+            let wallet = wallet::db::wallet::get_wallet(&db_conn, seed_phrase.clone())?;
+            if wallet.is_none() {
```

Comment:
on the contrary, if the wallet does already exist. Just return with a message saying that this wallet already exists